docs/sdk-upgrade.md

```diff
@@ -0,0 +1,176 @@
+
+Check NewRelic for errors during this time, also.
+
+#### For the next 1 hour
+
+After 15 minutes, per our [production deploy instructions](https://handbook.login.gov/articles/appdev-deploy.html#production), you may remove the old servers. This takes us back out from a 50/50 deployment state. These commands are for **production only**:
+
+```zsh
+aws-vault exec prod-power -- ./bin/scale-remove-old-instances prod idp
+aws-vault exec prod-power -- ./bin/scale-remove-old-instances prod worker
+```
+
+Manually check the document capture page in the environment you are deploying to. Ensure the SDK loads and can capture images.
+
+Check NewRelic for errors at the end of one hour.
+
+#### For 2 weeks after deploy:
+
+Monitor with this [AWS CloudWatch Acuant upgrade dashboard](https://us-west-2.console.aws.amazon.com/cloudwatch/home?region=us-west-2#dashboards:name=js-acuant-upgrade).
+
+![pie-charts-sdk](https://user-images.githubusercontent.com/546123/232889932-432e5cd5-c460-4a0a-8c6b-9f54324f327b.png)
+
+In this screenshot from the dashboard, the pie chart on the right shows a newer version of the SDK approaching 50% of document capture requests as A/B testing kicks in. The chart on the left shows that the newer version of the SDK is responsible for a proportionately lesser share of document capture failures, indicating that the new version is likely an improvement on the old.
+
+If the new version of the SDK is performing well for a couple weeks of A/B testing, it is time to cut over 100% of traffic to the new version per the next section.
+
+## Switch versions
+
+When the test is concluded, for better or for worse, you'll want to end A/B testing, leaving only one SDK version. The procedure is the same as the above procedure to [turn on A/B testing](#turn-on-ab-testing) with one exception:
+
+#### ✅ If the test went well
+
+The only line in the configuration file that needs to be changed is this:
+
+```yml
+idv_acuant_sdk_upgrade_a_b_testing_enabled: false
+```
+
+This switches A/B testing off. The new, desired SDK version should already be set in the `idv_acuant_sdk_version_default` field, so it should not need to be changed. Double-check that this number is correct.
+
+#### ❌ If the test went poorly
+
+If the new version under A/B testing is performing poorly, you may want to end the test, returning to the *old* SDK version. Hence, you should set the default back to the version you want. Edit these two lines of the configuration file:
+
+```yml
+idv_acuant_sdk_upgrade_a_b_testing_enabled: false
+idv_acuant_sdk_version_default: 11.M.M
+```
+
+#### 🔄 In either case
+Save the configuration file, recycle, and monitor the recycle just as you did when you turned A/B testing on.
+
+Once again, the command to recycle is:
+
+```zsh
+aws-vault exec prod-power -- bin/asg-recycle prod idp
+```
+Use the same [tools for monitoring](#monitor-ab-testing).
+
+### Cleanup
```

Comment:
There is one final step, which can go in this cleanup section or the preceding one: we have to set the all the correct default config values in `config/application.yml.default`, and then commit the changes to main. These defaults are what will be used by any environment -- including one's local dev environment -- when no explicit values are set in a config to override them.
  
The values in that file will need to be something like:
```yaml
    idv_acuant_sdk_upgrade_a_b_testing_enabled: false
    idv_acuant_sdk_upgrade_a_b_testing_percent: 50 # This is ignored, so won't matter
    idv_acuant_sdk_version_alternate: 11.M.<previous>
    idv_acuant_sdk_version_default: 11.N.<newest>
```

Reply:
Great point! I've added this to the instructions. I chose to only include the last two lines of `application.yml.default` because those are the parts that need to change as I understand it.